cli/daemon/run/testdata/echo/echo/config.go

```diff
@@ -0,0 +1,40 @@
+package echo
+
+import (
+	"context"
+
+	"encore.dev/config"
+)
+
+type CfgType struct {
+	ReadOnlyMode config.Bool
+	PublicKey    config.Bytes
+	AdminUsers   config.Values[string]
+
+	SubConfig config.Value[struct {
+		SubKey SubCfgType
+	}]
+}
+
+type SubCfgType struct {
+	MaxCount uint
+}
+
+var cfg = config.Load[CfgType]()
```

Comment:
How does this magic work? Why isn't this causing a compile error in the IDE?

Reply:
The public API will not have the two "hidden" arguments